compiler/src/main/scala/org/scalajs/core/compiler/PrepJSInterop.scala

```diff
@@ -574,10 +574,12 @@ abstract class PrepJSInterop extends plugins.PluginComponent
           for {
             annot <- sym.annotations
             annotSym = annot.symbol
-            if JSNativeLoadingSpecAnnots.contains(annotSym)
```

Comment:
If you move the guard to the body, consider also moving `annotSym` as a `val annotSym = ...` in the body of the for. It's a bit weird to have the last line of a for be an `=`.